File: mod/library/server/handle-message/exporter/_common/convert_to_hex.lua
local function convert_to_hex(colour_table)
	if not colour_table then return end
	local colour = "#" ..
		string.format("%02x", colour_table[1] * 255) ..
		string.format("%02x", colour_table[2] * 255) ..
		string.format("%02x", colour_table[3] * 255)
	return colour
end

return convert_to_hex

File: mod/library/server/handle-message/exporter/_common/convert-to-hex.ts
/**
 *
 * @param colourTable
 * @noSelf
 * @example
 */
const convertToHex = (colourTable: [number, number, number]) => {
	if (!colourTable) {
		return;
	}
	const [
		r,
		g,
		b
	] = colourTable;

	/**
	 *
	 * @param v
	 * @example
	 */
	const toHex = (v: number) => Math.round(v * 255)
		.toString(16)
		.padStart(2, "0");

	return `#${toHex(r)}${toHex(g)}${toHex(b)}`;
};

export default convertToHex;
